src/randomtrackingdict.py

```diff
@@ -108,8 +108,8 @@ def randomKeys(self, count=1):
         # pylint: disable=redefined-outer-name
         with self.lock:
             # reset if we've requested all
-            # or if last object received too long time ago
-            if self.pendingLen == self.len or self.lastObject + self.pendingTimeout > time():
+            # and if last object received too long time ago
+            if self.pendingLen == self.len and self.lastObject + self.pendingTimeout > time():
```

Comment:
I don't understand it, `self.lastObject + self.pendingTimeout > time()` means that timeout isn't over? The comment says the opposite.

Reply:
I think you're correct. I'll change it and retest.

Reply:
Retest looks ok.